src/pkg/exporter/system_collector.go

```diff
@@ -0,0 +1,86 @@
+package exporter
+
+import (
+	"encoding/json"
+
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// SystemInfoCollectorName ...
+const (
+	systemInfoCollectorName = "SystemInfoCollector"
+	sysInfoURL              = "/api/v2.0/systeminfo"
+)
+
+var (
+	harborSysInfo = typedDesc{
+		desc: newDescWithLables("", "system_info", "Information of Harbor system",
+			"auth_mode",
+			"registry_url",
+			"external_url",
+			"harbor_version",
+			"registry_storage_provider"),
+		valueType: prometheus.GaugeValue,
+	}
+)
+
+// NewSystemInfoCollector ...
+func NewSystemInfoCollector(hbrCli *HarborClient) *SystemInfoCollector {
+	return &SystemInfoCollector{
+		HarborClient: hbrCli,
+	}
+}
+
+// SystemInfoCollector ...
+type SystemInfoCollector struct {
+	*HarborClient
+}
+
+// Describe implements prometheus.Collector
+func (hc *SystemInfoCollector) Describe(c chan<- *prometheus.Desc) {
+	c <- harborSysInfo.Desc()
+}
+
+// Collect implements prometheus.Collector
+func (hc *SystemInfoCollector) Collect(c chan<- prometheus.Metric) {
+	for _, m := range hc.getSysInfo() {
+		c <- m
+	}
+}
+
+func (hc *SystemInfoCollector) getSysInfo() []prometheus.Metric {
+	if CacheEnabled() {
+		value, ok := CacheGet(systemInfoCollectorName)
+		if ok {
+			return value.([]prometheus.Metric)
+		}
+	}
+	result := []prometheus.Metric{}
+	res, err := hbrCli.Get(sysInfoURL)
+	if err != nil {
+		log.Errorf("request health info failed with err: %v", err)
+		return result
+	}
+	defer res.Body.Close()
+	var sysInfoResponse responseSysInfo
+	json.NewDecoder(res.Body).Decode(&sysInfoResponse)
+	result = append(result, harborSysInfo.MustNewConstMetric(1,
+		sysInfoResponse.AuthMode,
+		sysInfoResponse.RegistryURL,
+		sysInfoResponse.ExternalURL,
+		sysInfoResponse.HarborVersion,
+		sysInfoResponse.StorageProvider))
+	if CacheEnabled() {
+		CachePut(systemInfoCollectorName, result)
+	}
+	return result
+}
+
+type responseSysInfo struct {
```

Comment:
Reuse the data structure, this is defined in swagger and can be generated via `make gen_apis`

Also be careful that some attributes of the response will be missing the request is not authenticated:
https://github.com/goharbor/harbor/issues/9149